kubevirt start vm CrashLoopBackOff

<!-- Make sure that you visit our User Guide at https://kubevirt.io/user-guide.
-->

**What happened**:
create vm start error

**What you expected to happen**:
Normal startup vm

**How to reproduce it (as minimally and precisely as possible)**:
This error has a certain degree of contingency

**Additional context**:
pod logs：
{"component":"virt-launcher","level":"info","msg":"Collected all requested hook sidecar sockets","pos":"manager.go:86","timestamp":"2023-06-02T01:57:32.329210Z"}
{"component":"virt-launcher","level":"info","msg":"Sorted all collected sidecar sockets per hook point based on their priority and name: map[]","pos":"manager.go:89","timestamp":"2023-06-02T01:57:32.329266Z"}
{"component":"virt-launcher","level":"info","msg":"Connecting to libvirt daemon: qemu+unix:///session?socket=/var/run/libvirt/virtqemud-sock","pos":"libvirt.go:519","timestamp":"2023-06-02T01:57:32.329904Z"}
{"component":"virt-launcher","level":"info","msg":"Connecting to libvirt daemon failed: virError(Code=38, Domain=7, Message='Failed to connect socket to '/var/run/libvirt/virtqemud-sock': No such file or directory')","pos":"libvirt.go:527","timestamp":"2023-06-02T01:57:32.330622Z"}
{"component":"virt-launcher","level":"info","msg":"libvirt version: 9.0.0, package: 3.el9 (builder@centos.org, 2023-02-01-12:58:56, )","subcomponent":"libvirt","thread":"43","timestamp":"2023-06-02T01:57:32.348000Z"}
{"component":"virt-launcher","level":"info","msg":"hostname: demo0602","subcomponent":"libvirt","thread":"43","timestamp":"2023-06-02T01:57:32.348000Z"}
{"component":"virt-launcher","level":"error","msg":"internal error: Unable to get session bus connection: Cannot spawn a message bus without a machine-id: Unable to load /var/lib/dbus/machine-id or /etc/machine-id: Failed to open file “/var/lib/dbus/machine-id”: No such file or directory","pos":"virGDBusGetSessionBus:126","subcomponent":"libvirt","thread":"43","timestamp":"2023-06-02T01:57:32.348000Z"}
{"component":"virt-launcher","level":"error","msg":"internal error: Unable to get system bus connection: Could not connect: No such file or directory","pos":"virGDBusGetSystemBus:99","subcomponent":"libvirt","thread":"43","timestamp":"2023-06-02T01:57:32.348000Z"}
{"component":"virt-launcher","level":"info","msg":"Connected to libvirt daemon","pos":"libvirt.go:535","timestamp":"2023-06-02T01:57:32.832927Z"}
{"component":"virt-launcher","level":"info","msg":"Registered libvirt event notify callback","pos":"client.go:539","timestamp":"2023-06-02T01:57:32.835073Z"}
{"component":"virt-launcher","level":"info","msg":"Marked as ready","pos":"virt-launcher.go:75","timestamp":"2023-06-02T01:57:32.835244Z"}
{"component":"virt-launcher","kind":"","level":"info","msg":"Executing PreStartHook on VMI pod environment","name":"ed380a96-383f-40ec-9b70-ad64a4fb7297","namespace":"default","pos":"manager.go:531","timestamp":"2023-06-02T01:57:33.094210Z","uid":"b41434e8-a2b4-450d-b55a-291c94ae7045"}
{"component":"virt-launcher","kind":"","level":"info","msg":"Starting PreCloudInitIso hook","name":"ed380a96-383f-40ec-9b70-ad64a4fb7297","namespace":"default","pos":"manager.go:552","timestamp":"2023-06-02T01:57:33.094270Z","uid":"b41434e8-a2b4-450d-b55a-291c94ae7045"}
{"component":"virt-launcher","level":"info","msg":"Found nameservers in /etc/resolv.conf: \ufffd\u0014\u0000\u0010","pos":"network.go:287","timestamp":"2023-06-02T01:57:33.095518Z"}
{"component":"virt-launcher","level":"info","msg":"Found search domains in /etc/resolv.conf: default.svc.cluster.local svc.cluster.local cluster.local","pos":"network.go:288","timestamp":"2023-06-02T01:57:33.095550Z"}
{"component":"virt-launcher","level":"info","msg":"Starting SingleClientDHCPServer","pos":"server.go:65","timestamp":"2023-06-02T01:57:33.095591Z"}
{"component":"virt-launcher","level":"info","msg":"Driver cache mode for /dev/root-0 set to none","pos":"converter.go:468","timestamp":"2023-06-02T01:57:33.095653Z"}
{"component":"virt-launcher","level":"info","msg":"Driver IO mode for /dev/root-0 set to native","pos":"converter.go:509","timestamp":"2023-06-02T01:57:33.095675Z"}
{"component":"virt-launcher","level":"info","msg":"Driver cache mode for /var/run/kubevirt-ephemeral-disks/cloud-init-data/default/ed380a96-383f-40ec-9b70-ad64a4fb7297/noCloud.iso set to none","pos":"converter.go:468","timestamp":"2023-06-02T01:57:33.095834Z"}
{"component":"virt-launcher","level":"warning","msg":"Failed to get image info","pos":"manager.go:715","reason":"failed to invoke qemu-img: exit status 1","timestamp":"2023-06-02T01:57:33.103073Z"}
{"component":"virt-launcher","kind":"","level":"info","msg":"Domain XML generated. Base64 dump PGRvbWFpbiB0eXBlPSJrdm0iIHhtbG5zOnFlbXU9Imh0dHA6Ly9saWJ2aXJ0Lm9yZy9zY2hlbWFzL2RvbWFpbi9xZW11LzEuMCI+Cgk8bmFtZT5kZWZhdWx0X2VkMzgwYTk2LTM4M2YtNDBlYy05YjcwLWFkNjRhNGZiNzI5NzwvbmFtZT4KCTxtZW1vcnkgdW5pdD0iYiI+ODA2NTY0NjU5MjwvbWVtb3J5PgoJPG9zPgoJCTx0eXBlIGFyY2g9Ing4Nl82NCIgbWFjaGluZT0icTM1Ij5odm08L3R5cGU+CgkJPHNtYmlvcyBtb2RlPSJzeXNpbmZvIj48L3NtYmlvcz4KCTwvb3M+Cgk8c3lzaW5mbyB0eXBlPSJzbWJpb3MiPgoJCTxzeXN0ZW0+CgkJCTxlbnRyeSBuYW1lPSJ1dWlkIj5lZDM4MGE5Ni0zODNmLTQwZWMtOWI3MC1hZDY0YTRmYjcyOTc8L2VudHJ5PgoJCQk8ZW50cnkgbmFtZT0ibWFudWZhY3R1cmVyIj5LdWJlVmlydDwvZW50cnk+CgkJCTxlbnRyeSBuYW1lPSJmYW1pbHkiPkt1YmVWaXJ0PC9lbnRyeT4KCQkJPGVudHJ5IG5hbWU9InByb2R1Y3QiPk5vbmU8L2VudHJ5PgoJCQk8ZW50cnkgbmFtZT0ic2t1Ij48L2VudHJ5PgoJCQk8ZW50cnkgbmFtZT0idmVyc2lvbiI+PC9lbnRyeT4KCQk8L3N5c3RlbT4KCQk8Ymlvcz48L2Jpb3M+CgkJPGJhc2VCb2FyZD48L2Jhc2VCb2FyZD4KCQk8Y2hhc3Npcz48L2NoYXNzaXM+Cgk8L3N5c2luZm8+Cgk8ZGV2aWNlcz4KCQk8aW50ZXJmYWNlIHR5cGU9ImV0aGVybmV0Ij4KCQkJPHNvdXJjZT48L3NvdXJjZT4KCQkJPHRhcmdldCBkZXY9InRhcDAiIG1hbmFnZWQ9Im5vIj48L3RhcmdldD4KCQkJPG1vZGVsIHR5cGU9InZpcnRpby1ub24tdHJhbnNpdGlvbmFsIj48L21vZGVsPgoJCQk8bWFjIGFkZHJlc3M9IjUyOjU0OjAwOjc4OmYzOjQ4Ij48L21hYz4KCQkJPG10dSBzaXplPSIxNTAwIj48L210dT4KCQkJPGFsaWFzIG5hbWU9InVhLWV0aDAiPjwvYWxpYXM+CgkJCTxyb20gZW5hYmxlZD0ibm8iPjwvcm9tPgoJCTwvaW50ZXJmYWNlPgoJCTxjaGFubmVsIHR5cGU9InVuaXgiPgoJCQk8dGFyZ2V0IG5hbWU9Im9yZy5xZW11Lmd1ZXN0X2FnZW50LjAiIHR5cGU9InZpcnRpbyI+PC90YXJnZXQ+CgkJPC9jaGFubmVsPgoJCTxjb250cm9sbGVyIHR5cGU9InVzYiIgaW5kZXg9IjAiIG1vZGVsPSJxZW11LXhoY2kiPjwvY29udHJvbGxlcj4KCQk8Y29udHJvbGxlciB0eXBlPSJzY3NpIiBpbmRleD0iMCIgbW9kZWw9InZpcnRpby1ub24tdHJhbnNpdGlvbmFsIj48L2NvbnRyb2xsZXI+CgkJPGNvbnRyb2xsZXIgdHlwZT0idmlydGlvLXNlcmlhbCIgaW5kZXg9IjAiIG1vZGVsPSJ2aXJ0aW8tbm9uLXRyYW5zaXRpb25hbCI+PC9jb250cm9sbGVyPgoJCTx2aWRlbz4KCQkJPG1vZGVsIHR5cGU9InZnYSIgaGVhZHM9IjEiIHZyYW09IjE2Mzg0Ij48L21vZGVsPgoJCTwvdmlkZW8+CgkJPGdyYXBoaWNzIHR5cGU9InZuYyI+CgkJCTxsaXN0ZW4gdHlwZT0ic29ja2V0IiBzb2NrZXQ9Ii92YXIvcnVuL2t1YmV2aXJ0LXByaXZhdGUvYjQxNDM0ZTgtYTJiNC00NTBkLWI1NWEtMjkxYzk0YWU3MDQ1L3ZpcnQtdm5jIj48L2xpc3Rlbj4KCQk8L2dyYXBoaWNzPgoJCTxtZW1iYWxsb29uIG1vZGVsPSJ2aXJ0aW8tbm9uLXRyYW5zaXRpb25hbCIgZnJlZVBhZ2VSZXBvcnRpbmc9Im9uIj4KCQkJPHN0YXRzIHBlcmlvZD0iMTAiPjwvc3RhdHM+CgkJPC9tZW1iYWxsb29uPgoJCTxkaXNrIGRldmljZT0iZGlzayIgdHlwZT0iYmxvY2siIG1vZGVsPSJ2aXJ0aW8tbm9uLXRyYW5zaXRpb25hbCI+CgkJCTxzb3VyY2UgZGV2PSIvZGV2L3Jvb3QtMCIgbmFtZT0icm9vdC0wIj48L3NvdXJjZT4KCQkJPHRhcmdldCBidXM9InZpcnRpbyIgZGV2PSJ2ZGEiPjwvdGFyZ2V0PgoJCQk8ZHJpdmVyIGNhY2hlPSJub25lIiBlcnJvcl9wb2xpY3k9InN0b3AiIGlvPSJuYXRpdmUiIG5hbWU9InFlbXUiIHR5cGU9InJhdyIgZGlzY2FyZD0idW5tYXAiPjwvZHJpdmVyPgoJCQk8YWxpYXMgbmFtZT0idWEtcm9vdC0wIj48L2FsaWFzPgoJCQk8Ym9vdCBvcmRlcj0iMSI+PC9ib290PgoJCQk8ZmlsZXN5c3RlbU92ZXJoZWFkPjA8L2ZpbGVzeXN0ZW1PdmVyaGVhZD4KCQkJPGNhcGFjaXR5PjIxNDc0ODM2NDgwPC9jYXBhY2l0eT4KCQkJPGV4cGFuZERpc2tzRW5hYmxlZD50cnVlPC9leHBhbmREaXNrc0VuYWJsZWQ+CgkJPC9kaXNrPgoJCTxkaXNrIGRldmljZT0iZGlzayIgdHlwZT0iZmlsZSIgbW9kZWw9InZpcnRpby1ub24tdHJhbnNpdGlvbmFsIj4KCQkJPHNvdXJjZSBmaWxlPSIvdmFyL3J1bi9rdWJldmlydC1lcGhlbWVyYWwtZGlza3MvY2xvdWQtaW5pdC1kYXRhL2RlZmF1bHQvZWQzODBhOTYtMzgzZi00MGVjLTliNzAtYWQ2NGE0ZmI3Mjk3L25vQ2xvdWQuaXNvIj48L3NvdXJjZT4KCQkJPHRhcmdldCBidXM9InZpcnRpbyIgZGV2PSJ2ZGIiPjwvdGFyZ2V0PgoJCQk8ZHJpdmVyIGNhY2hlPSJub25lIiBlcnJvcl9wb2xpY3k9InN0b3AiIG5hbWU9InFlbXUiIHR5cGU9InJhdyIgZGlzY2FyZD0idW5tYXAiPjwvZHJpdmVyPgoJCQk8YWxpYXMgbmFtZT0idWEtY2xvdWRpbml0ZGlzayI+PC9hbGlhcz4KCQkJPGV4cGFuZERpc2tzRW5hYmxlZD50cnVlPC9leHBhbmREaXNrc0VuYWJsZWQ+CgkJPC9kaXNrPgoJCTxpbnB1dCB0eXBlPSJ0YWJsZXQiIGJ1cz0idXNiIj4KCQkJPGFsaWFzIG5hbWU9InVhLXRhYmxldCI+PC9hbGlhcz4KCQk8L2lucHV0PgoJCTxzZXJpYWwgdHlwZT0idW5peCI+CgkJCTx0YXJnZXQgcG9ydD0iMCI+PC90YXJnZXQ+CgkJCTxzb3VyY2UgbW9kZT0iYmluZCIgcGF0aD0iL3Zhci9ydW4va3ViZXZpcnQtcHJpdmF0ZS9iNDE0MzRlOC1hMmI0LTQ1MGQtYjU1YS0yOTFjOTRhZTcwNDUvdmlydC1zZXJpYWwwIj48L3NvdXJjZT4KCQk8L3NlcmlhbD4KCQk8Y29uc29sZSB0eXBlPSJwdHkiPgoJCQk8dGFyZ2V0IHR5cGU9InNlcmlhbCIgcG9ydD0iMCI+PC90YXJnZXQ+CgkJPC9jb25zb2xlPgoJCTxybmcgbW9kZWw9InZpcnRpby1ub24tdHJhbnNpdGlvbmFsIj4KCQkJPGJhY2tlbmQgbW9kZWw9InJhbmRvbSI+L2Rldi91cmFuZG9tPC9iYWNrZW5kPgoJCTwvcm5nPgoJCTxzb3VuZCBtb2RlbD0iaWNoOSI+CgkJCTxhbGlhcyBuYW1lPSJ1YS1teS1zb3VuZCI+PC9hbGlhcz4KCQk8L3NvdW5kPgoJPC9kZXZpY2VzPgoJPGNsb2NrIG9mZnNldD0idXRjIiBhZGp1c3RtZW50PSIyODgwMCI+PC9jbG9jaz4KCTxtZXRhZGF0YT4KCQk8a3ViZXZpcnQgeG1sbnM9Imh0dHA6Ly9rdWJldmlydC5pbyI+CgkJCTx1aWQ+PC91aWQ+CgkJPC9rdWJldmlydD4KCTwvbWV0YWRhdGE+Cgk8ZmVhdHVyZXM+CgkJPGFjcGk+PC9hY3BpPgoJPC9mZWF0dXJlcz4KCTxjcHUgbW9kZT0iaG9zdC1tb2RlbCI+CgkJPHRvcG9sb2d5IHNvY2tldHM9IjEiIGNvcmVzPSI0IiB0aHJlYWRzPSIxIj48L3RvcG9sb2d5PgoJPC9jcHU+Cgk8dmNwdSBwbGFjZW1lbnQ9InN0YXRpYyI+NDwvdmNwdT4KCTxpb3RocmVhZHM+MTwvaW90aHJlYWRzPgo8L2RvbWFpbj4=","name":"ed380a96-383f-40ec-9b70-ad64a4fb7297","namespace":"default","pos":"libvirt_helper.go:133","timestamp":"2023-06-02T01:57:33.103999Z","uid":"b41434e8-a2b4-450d-b55a-291c94ae7045"}
{"component":"virt-launcher-monitor","level":"info","msg":"Reaped pid 57 with status 0","pos":"virt-launcher-monitor.go:125","timestamp":"2023-06-02T01:57:33.144901Z"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x15a5d90]

goroutine 63 [running]:
kubevirt.io/kubevirt/pkg/virt-launcher/notify-client.(*Notifier).StartDomainNotifier.func1()
        pkg/virt-launcher/notify-client/client.go:444 +0x810
created by kubevirt.io/kubevirt/pkg/virt-launcher/notify-client.(*Notifier).StartDomainNotifier
        pkg/virt-launcher/notify-client/client.go:413 +0x2ea
{"component":"virt-launcher-monitor","level":"info","msg":"Reaped pid 12 with status 512","pos":"virt-launcher-monitor.go:125","timestamp":"2023-06-02T01:57:33.198021Z"}
{"component":"virt-launcher-monitor","level":"error","msg":"dirty virt-launcher shutdown: exit-code 2","pos":"virt-launcher-monitor.go:143","timestamp":"2023-06-02T01:57:33.198085Z"}
{"component":"virt-launcher-monitor","level":"info","msg":"Reaped pid 59 with status 0","pos":"virt-launcher-monitor.go:125","timestamp":"2023-06-02T01:57:33.203177Z"}


**Environment**:
- KubeVirt version (use `virtctl version`): v1.0.0-beta.0
- Kubernetes version (use `kubectl version`): v1.23.8
- VM or VMI specifications: 
- Cloud provider or hardware configuration: N/A
- OS (e.g. from /etc/os-release):  CentOS Linux 7
- Kernel (e.g. `uname -a`): Linux node1 5.4.207-1.el7.elrepo.x86_64
- Install tools: N/A
- Others: N/A

--- a/pkg/virt-launcher/notify-client/client.go
+++ b/pkg/virt-launcher/notify-client/client.go
@@ -440,22 +440,26 @@ func (n *Notifier) StartDomainNotifier(
 				n.SendDomainEvent(newWatchEventError(fmt.Errorf("Libvirt reconnect, domain %s", domainName)))
 
 			case <-metadataCache.Listen():
-				domainCache = util.NewDomainFromName(
-					util.DomainFromNamespaceName(domainCache.ObjectMeta.Namespace, domainCache.ObjectMeta.Name),
-					vmi.UID,
-				)
-				eventCallback(
-					domainConn,
-					domainCache,
-					libvirtEvent{},
-					n,
-					deleteNotificationSent,
-					interfaceStatuses,
-					guestOsInfo,
-					vmi,
-					fsFreezeStatus,
-					metadataCache,
-				)
+				// Metadata cache updates should be processed only *after* at least one
+				// libvirt event arrived (which creates the first domainCache).
+				if domainCache != nil {
+					domainCache = util.NewDomainFromName(
+						util.DomainFromNamespaceName(domainCache.ObjectMeta.Namespace, domainCache.ObjectMeta.Name),
+						vmi.UID,
+					)
+					eventCallback(
+						domainConn,
+						domainCache,
+						libvirtEvent{},
+						n,
+						deleteNotificationSent,
+						interfaceStatuses,
+						guestOsInfo,
+						vmi,
+						fsFreezeStatus,
+						metadataCache,
+					)
+				}
 			}
 		}
 	}()

